servlet/src/main/java/io/undertow/servlet/handlers/ServletPathMatches.java

```diff
@@ -64,12 +68,49 @@
 
     private volatile ServletPathMatchesData data;
 
-    private final LRUCache<String, ServletPathMatch> pathMatchCache = new LRUCache<>(1000, -1, true); //TODO: configurable
+    private static final int CACHE_SIZE = Integer.getInteger("io.undertow.servlet.path.matches.cacheSize", 1000);
```

Comment:
Ditto for the two Jiras, here we are adding support to setting cache size, it is not the bug fix.

Reply:
Yes, yes, I just added this because it makes sense. This can be removed for this JIRA. :-)

Reply:
So, as we discussed in the chat, let's split this in two. As soon as the PR is updated I'll merge it, it will go in the next release. Thanks a lot for your PR btw!

Reply:
OK, removed the CACHE_SIZE from this PR!